### PR TITLE
Stats schema update/v5

### PR DESCRIPTION
**Pull request by inashivb**
Description of some stats params.
Can be merged as-is once approved or we can wait for other stats updates too.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/6434

Previous PR: https://github.com/OISF/suricata/pull/10324

Changes since v3:
- Moved description above type even in refs case
- added description to refs in applayer errors
- added description to applayer.flow stats

### CODE REVIEW

File: etc/schema.json [hunk 1]
```diff
@@ -3669,126 +3669,165 @@
...
                 },
                 "memcap_pressure_max": {
+                    "description": "Maximum memcap_pressure seen by the engine",
                     "type": "integer"
                 },
                 "app_layer": {
                     "type": "object",
                     "properties": {
                         "expectations": {
+                            "description": "Expectation (dynamic parallel flow) counter",
```
Comment (victorjulien) [inline etc/schema.json:3687]: what is 'dynamic parallel flow'? Its not something we refer to anywhere else?

File: etc/schema.json [hunk 2]
```diff
@@ -3669,126 +3669,165 @@
...
                                 "ftp-data": {
+                                    "description": "Errors encountered parsing FTP data",
                                     "$ref": "#/$defs/stats_applayer_error"
                                 },
                                 "http": {
+                                    "description": "Errors encountered parsing HTTP",
                                     "$ref": "#/$defs/stats_applayer_error"
                                 },
                                 "http2": {
+                                    "description": "Errors encountered parsing HTTP 2",
```
Comment (victorjulien) [inline etc/schema.json:3746]: generally referred to as `HTTP/2`

File: etc/schema.json [hunk 3]
```diff
@@ -3669,126 +3669,165 @@
...
                                 "ike": {
+                                    "description": "Errors encountered parsing IKE protocol",
                                     "$ref": "#/$defs/stats_applayer_error"
                                 },
                                 "imap": {
+                                    "description": "Errors encountered parsing IMAP",
                                     "$ref": "#/$defs/stats_applayer_error"
                                 },
                                 "krb5_tcp": {
+                                    "description": "Errors encountered parsing Kerberos5/TCP protocol",
```
Comment (victorjulien) [inline etc/schema.json:3758]: Kerberos version 5? It's version 5 of the protocol.

File: etc/schema.json [hunk 4]
```diff
@@ -3669,126 +3669,165 @@
...
                                 "nfs_tcp": {
+                                    "description": "Errors encountered parsing NFS/TCP protocol",
                                     "$ref": "#/$defs/stats_applayer_error"
                                 },
                                 "nfs_udp": {
+                                    "description": "Errors encountered parsing NFS/UDP protocol",
                                     "$ref": "#/$defs/stats_applayer_error"
                                 },
                                 "ntp": {
+                                    "description": "Errors encountered parsing NTP",
```
Comment (victorjulien) [inline etc/schema.json:3782]: most entries have "<proto> _protocol_", would be good to be consistent

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: etc/schema.json (hunk 4 above)
Comment (inashivb) [reply etc/schema.json:3782]: I skipped it in places where the protocol's abbreviation ended in `P` that stood for protocol. Like here. Else it becomes _Network Time Protocol_ protocol.
If you think adding a protocol is more convenient to read, lmk.

Comment (victorjulien) [reply etc/schema.json:3782]: ah, makes sense now, thanks

**Pull request closed without merging**